### PR TITLE
Entrances support for bash executor

### PR DESCRIPTION
**Pull request by roleyfoley**
## Description
Updates to align with https://github.com/hamlet-io/engine/pull/1407 and broader support for deployment groups 

- Adds entrance configuration to provide the command line option for the entrance you want to use 
- removes isValidUnit check as the unit and group is now part of the hamlet engine and can't be determined in the executor 
- Sets the single entry point to the hamlet engine since we now have entrances to define different ways of generating outputs 
- Allows for dynamic levels and entrance document types 
    - If an entrance or level doesn't match one of our existing ones, files will be created under `"${PRODUCT_STATE_DIR}/cot/${ENVIRONMENT}/${SEGMENT}"` directory with the entrance as the first prefix to the file name 

## Motivation and Context
Allows for control over deployment units, groups and entrances to hamlet to be handled inside of the hamlet engine

## How Has This Been Tested?
Tested locally

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] Refactor (non-breaking change which improves the structure or operation of the implementation)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Followup Actions
- [ ] engine - https://github.com/hamlet-io/engine/pull/1407
- [ ] executor-bash - https://github.com/hamlet-io/executor-bash/pull/105
- [ ] engine-plugin-aws - https://github.com/hamlet-io/engine-plugin-aws/pull/132
- [ ] engine-plugin-azure - https://github.com/hamlet-io/engine-plugin-azure/pull/204 

## Checklist:
- [ ] My change requires a change to the [documentation](https://github.com/hamlet-io/docs).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [X] None of the above.

### CODE REVIEW

**Review by ml019 — COMMENTED** [review]
Does model need to be removed too as it is mostly controlled by the entrance? Would we ever want to vary the model/flow independent of the entrance?

If not, should flow be a config parameter?

**Comment by roleyfoley** [comment]
Added flow and removed model

File: cli/createTemplate.sh [hunk 1]
```diff
@@ -25,6 +27,7 @@ Usage: $(basename $0) -l LEVEL -u DEPLOYMENT_UNIT -c CONFIGURATION_REFERENCE -q
 
 where
 
+(o) -b FLOW                    is a flow through hamlet you want to invoke to perform your task
```
Comment (ml019) [inline cli/createTemplate.sh:30]: update to reflect multiple flows

File: cli/createTemplate.sh [hunk 2]
```diff
@@ -53,22 +56,21 @@ GENERATION_FRAMEWORK    = "${GENERATION_FRAMEWORK_DEFAULT}"
...
 EOF
 }
 
 function options() {
 
   # Parse options
-  while getopts ":c:d:e:f:g:hi:l:o:p:q:r:u:xz:" option; do
+  while getopts ":b:c:d:e:f:g:hi:l:o:p:q:r:u:xz:" option; do
       case "${option}" in
+          b) FLOWS+=("${OPTARG}") ;;
```
Comment (ml019) [inline cli/createTemplate.sh:73]: Need to initialise before getopts?

**Review by ml019 — CHANGES_REQUESTED** [review]

File: cli/createTemplate.sh (hunk 1 above)
Comment (roleyfoley) [reply cli/createTemplate.sh:30]: You can provide the -b option multiple times, so each parameter is one flow

File: cli/createTemplate.sh (hunk 2 above)
Comment (roleyfoley) [reply cli/createTemplate.sh:73]: This is setup on Line 19 during the setup process

**Pull request merged**